backend/api/src/main.rs

```diff
@@ -18,7 +19,10 @@ async fn main() -> Result<()> {
 
     tracing::info!("starting");
 
-    api::run(global).await?;
+    select! {
+        _ = api::run(global.clone()) => tracing::info!("api stopped"),
+        _ = tokio::signal::ctrl_c() => tracing::info!("ctrl-c received"),
+    }
```

Comment:
Does this allow for graceful shutdown?
How does `Future` cancelling even work in Rust? The page for `Future`[^1] doesn't mention cancellation at all. The `select!` page[^2] broadly explains what "cancellation safety" means:
> when an asynchronous method is cancelled, that always happens at an `.await`. If your function behaves correctly even if it is restarted while waiting at an `.await`, then it is cancellation safe.

Going off that explanation we should be careful about what we do in the `api::run` function and should guarantee that any `.await` is a "stable" point in the program that it can safely exit out at and never return to.

Some other interesting links include a blog post about Rust `Future`s in general and some ideas about the shiny future (posted mid 2021)[^3] and another blog post about specifically Rust `Future` cancellation[^4].

[^1]: https://doc.rust-lang.org/std/future/trait.Future.html
[^2]: https://docs.rs/tokio/1.25.0/tokio/macro.select.html
[^3]: https://carllerche.com/2021/06/17/six-ways-to-make-async-rust-easier/
[^4]: https://blog.yoshuawuyts.com/async-cancellation-1/

Reply:
It doesnt allow for graceful shutdown, tokio does not have true structured concurrency if we want that we will have to implement it.

Reply:
Out of the scope of this PR, since this PR was focused on docker images and i wanted the image to shutdown when docker sends a shutdown signal